### ISSUE TITLE
Network.close() implicitly declared as throws Exception

### ISSUE THREAD

**Issue by KantarBenedictAdamson** [testcontainers/testcontainers-java#1389]
The `Network` interface simply declares itself as `extends AutoCloseable`. It therefore inherits the `throws` clause from `AutoCloseable`, which is `throws Exception`. That is, it is implicitly declared as throwing a **checked** exception, which is inconvenient (but sometimes necessary). But as the *actual* implementation in `Network.NetworkImpl` *does not* throw any checked exceptions (`Network.NetworkImpl.close()` has no `throws` clause), this is an unnecessary inconvenience. Leaving the method as implicitly `throws Exception` is also against the advice of the [`AutoCloseable.close()`](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html#close--) specification:

> While this interface method is declared to throw `Exception`, implementers are *strongly* encouraged to declare concrete implementations of the close method to throw more specific exceptions, or to throw no exception at all if the close operation cannot fail.

(A minor annoyance)

**Comment by KantarBenedictAdamson** [comment]
The fix is, of course, simply:
```
public interface Network extends AutoCloseable, TestRule {
    @Override
    void close();

...
```

**Comment by bsideup** [comment]
Hi @KantarBenedictAdamson,

Would you like to contribute it?

**Comment by KantarBenedictAdamson** [comment]
I'm not sure I can contribute yet; I'm looking into using Testcontainers at my workplace. If the demonstration is OK, I'll then talk to my boss about making OSS contributions. Sorry I can't do more now. :-(

**Comment by rnorth** [comment]
Released in [1.11.2](https://github.com/testcontainers/testcontainers-java/releases/tag/1.11.2)!
